crates/macro-support/src/parser.rs

```diff
@@ -1568,10 +1577,22 @@ impl MacroParse<&ClassMarker> for &mut syn::ImplItemFn {
             let kind = operation_kind(&opts);
             ast::MethodKind::Operation(ast::Operation { is_static, kind })
         };
+
+        // Validate that js_namespace is not used on methods
+        if let Some((_, span)) = opts.js_namespace() {
+            return Err(Diagnostic::span_error(
+                span[0],
+                "`js_namespace` cannot be used on methods, getters, setters, or static methods. \
+                Use `js_namespace` on the exported struct definition instead to put the entire class in a namespace.",
+            ));
+        }
+
+        let js_namespace = opts.js_namespace().map(|(ns, _)| ns.0);
```

Comment:
Didn't we just check that `opts.namespace()` is `None`? So won't we always set `js_namespace` to `None` here?

Reply:
Yes exactly, added per below.